packages/gatsby/src/utils/parcel/__tests__/compile-gatsby-files.ts

```diff
@@ -175,6 +176,37 @@ describe(`gatsby file compilation`, () => {
       })
     })
   })
+
+  it(`handles errors in TS code`, async () => {
+    process.chdir(dir.errorInCode)
+    await remove(`${dir.errorInCode}/.cache`)
+    await compileGatsbyFiles(dir.errorInCode)
+
+    expect(reporterPanicMock).toMatchInlineSnapshot(`
+      [MockFunction] {
+        "calls": Array [
+          Array [
+            Object {
+              "context": Object {
+                "filePath": "<PROJECT_ROOT>/gatsby-node.ts",
+                "generalMessage": "Expected ';', '}' or <eof>",
+                "hints": null,
+                "origin": "@parcel/transformer-js",
+                "specificMessage": "This is the expression part of an expression statement",
```

Comment:
there is improvements to be made here to point out specific location as error we get is like this:
```
{
      diagnostics: [
        {
          message: "Expected ';', '}' or <eof>",
          codeFrames: [
            {
              filePath: '/Users/misiek/dev/pgs-gatsby/packages/gatsby/src/utils/parcel/__tests__/fixtures/error-in-code-ts/gatsby-node.ts',
              codeHighlights: [
                {
                  message: 'This is the expression part of an expression statement',
                  start: { line: 5, column: 1 },
                  end: { line: 5, column: 4 }
                },
                {
                  message: null,
                  start: { line: 5, column: 6 },
                  end: { line: 5, column: 7 }
                }
              ]
            }
          ],
          hints: null,
          origin: '@parcel/transformer-js'
        }
      ]
    }
```
but it's out of scope of this PR - purpose here is to try to handle segfaults without introducing regression - this added tests is just just a test for case we didn't have (what's the behavior on ts files that are invalid) and that I did break before my most recent commit